feat: send Slack message to #project-blt-bacon on new bacon claim (#3754)

mahii-17 opened the pull request:
### What this PR does

Adds Slack integration to notify the `#project-blt-bacon` channel whenever a new Bacon claim is submitted via the claim portal.

### 💬 Message format:
 New bacon claim submitted by <username> 🔗 <GitHub PR link> 💸 Bacon Requested: <amount>

### ✅ Implementation details

- Uses `slack_sdk.WebClient` to send messages
- Token loaded securely from `.env` via `os.getenv("SLACK_BOT_TOKEN")`
- `.env` ignored by Git; `.env.local` used locally for secrets
- Slack message sent in `BaconSubmissionView` after a successful submission
- Error handling added to avoid Slack errors breaking the flow
- Implements org-based Slack routing for bacon claims.
Closes  #3754


### 📄 Environment setup

Add this to your `.env` or `.env.local` file:

```env
SLACK_BOT_TOKEN=xoxb-your-real-slack-token
Slack bot must be added to the #project-blt-bacon channel
Required scope: chat:write
Additional cleanup
Ran black and isort for code formatting
Confirmed .env tracking is safe and clean

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Added Slack messaging notifications for new submissions, with channels determined by organization.
  - Enabled organization information capture for submissions.
  - Introduced support for multiple video hosting platforms (YouTube and Vimeo) for lecture URLs.
  - Added new environment variables for Slack API and PostgreSQL configuration.

- **Chores**
  - Updated configuration settings and dependency versions to support Slack integration and database connectivity.

- **Refactor/Style**
  - Improved code readability via enhanced formatting and clearer in-line comments.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

mahii-17 committed:
feat: send Slack message to #project-blt-bacon on new bacon claim

cicada0007 committed:
Added Security Incidents Summary to Dashboard  (#4029)

* added Security Issue

* fix

* fix isort

* ruff format fix

* ruff fix

* fix

DonnieBLT committed:
Update .gitignore

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

DonnieBLT committed:
Update .pre-commit-config.yaml

DonnieBLT reviewed (changes requested):
please use the org slack settings to send the message, and also add the org selection on the bacon claim

rinkitadhana committed:
Feat: Added Style Guide Page. (#4039)

* style-guide page completed

* nav-link added

* icon section added

* pre-commit error

igennova committed:
done (#4040)

dependabot[bot] committed:
chore(deps): Bump sentry-sdk from 2.23.1 to 2.24.0

Bumps [sentry-sdk](https://github.com/getsentry/sentry-python) from 2.23.1 to 2.24.0.
- [Release notes](https://github.com/getsentry/sentry-python/releases)
- [Changelog](https://github.com/getsentry/sentry-python/blob/master/CHANGELOG.md)
- [Commits](https://github.com/getsentry/sentry-python/compare/2.23.1...2.24.0)

---
updated-dependencies:
- dependency-name: sentry-sdk
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps-dev): Bump ruff from 0.11.1 to 0.11.2

Bumps [ruff](https://github.com/astral-sh/ruff) from 0.11.1 to 0.11.2.
- [Release notes](https://github.com/astral-sh/ruff/releases)
- [Changelog](https://github.com/astral-sh/ruff/blob/main/CHANGELOG.md)
- [Commits](https://github.com/astral-sh/ruff/compare/0.11.1...0.11.2)

---
updated-dependencies:
- dependency-name: ruff
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps): Bump openai from 1.67.0 to 1.68.2

Bumps [openai](https://github.com/openai/openai-python) from 1.67.0 to 1.68.2.
- [Release notes](https://github.com/openai/openai-python/releases)
- [Changelog](https://github.com/openai/openai-python/blob/main/CHANGELOG.md)
- [Commits](https://github.com/openai/openai-python/compare/v1.67.0...v1.68.2)

---
updated-dependencies:
- dependency-name: openai
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps-dev): Bump selenium from 4.29.0 to 4.30.0

Bumps [selenium](https://github.com/SeleniumHQ/Selenium) from 4.29.0 to 4.30.0.
- [Release notes](https://github.com/SeleniumHQ/Selenium/releases)
- [Commits](https://github.com/SeleniumHQ/Selenium/compare/selenium-4.29.0...selenium-4.30.0)

---
updated-dependencies:
- dependency-name: selenium
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

swaparup36 committed:
fixed /template_list too long to load (#4017)

dependabot[bot] committed:
chore(deps): Bump sentry-sdk from 2.24.0 to 2.24.1

Bumps [sentry-sdk](https://github.com/getsentry/sentry-python) from 2.24.0 to 2.24.1.
- [Release notes](https://github.com/getsentry/sentry-python/releases)
- [Changelog](https://github.com/getsentry/sentry-python/blob/master/CHANGELOG.md)
- [Commits](https://github.com/getsentry/sentry-python/compare/2.24.0...2.24.1)

---
updated-dependencies:
- dependency-name: sentry-sdk
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

DonnieBLT committed:
Update enforce-issue-number-in-description.yml

dependabot[bot] committed:
chore(deps): Bump python-dotenv from 1.0.1 to 1.1.0

Bumps [python-dotenv](https://github.com/theskumar/python-dotenv) from 1.0.1 to 1.1.0.
- [Release notes](https://github.com/theskumar/python-dotenv/releases)
- [Changelog](https://github.com/theskumar/python-dotenv/blob/main/CHANGELOG.md)
- [Commits](https://github.com/theskumar/python-dotenv/compare/v1.0.1...v1.1.0)

---
updated-dependencies:
- dependency-name: python-dotenv
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

SahilDhillon21 committed:
Merge commit from fork

* Fix vuln

* Fix edit lecture view

* Properly validate youtue/vimeo urls

* Add proper redirect

gojo-satorou-v7 committed:
Some fixes for organization dashboard. (#4059)

* Fixed url sanitization and removed redundant ssrf check

* Added username validation

* accidentally pushed some old changes

* removed logo check again

swaparup36 committed:
replaced /social code with content from queue model (#4053)

* replaced /social code with content from queue model

* paginator added to /social

* done the changes suggested by bot

* added filter for only launched items in queue model

igennova committed:
Now Profanity check is available for Bug_reporting (#4060)

* done

* done

* Update website/views/issue.py

* Update website/views/issue.py

* done

* done

---------

Co-authored-by: DonnieBLT <128622481+DonnieBLT@users.noreply.github.com>

mahii-17 committed:
feat: add organization-based Slack routing and update view logic

mahii-17 committed:
chore: commit remaining changes before merging upstream

mahii-17 committed:
chore: finalize merge conflict resolution and dependency updates

mahii-17 committed:
Fix: update Docker env + poetry lock for Slack integration PR

mahii-17 committed:
Merge branch 'main' into feature/slack-on-claim

mahii-17 committed:
fix: resolve CodeRabbit review comments (pagination, imports, URL validation)

mahii-17 committed:
fix(docker): add missing build tools and system dependencies

mahii-17 committed:
chore: sync poetry.lock with pyproject.toml for CI build

mahii-17 committed:
Merge branch 'main' into feature/slack-on-claim
